test/no_this_in_dyncall.js

```diff
@@ -0,0 +1,15 @@
+mergeInto(LibraryManager.library, {
+	$classLike: {
+		fnptr: 0,
+    call: function(val) {
+      {{{ makeDynCall('vp', 'this.fnptr') }}}(val);
+    }
+	},
+
+  test__deps: ['$classLike'],
+  test: function(fnptr, val) {
```

Comment:
The way we approach this issue with MEMORY64 is to tag all pointer-sized argument with `p` in the `__sig`.

So this `test` function should have a corresponding `test_sig: "pi"... then the value will be automatically converted to a Number/i53 as soon as it arrives on the JS side.  That way all JS code and just assume that pointers are always numbers and never bitint.

Given that, I suggest we perhaps revert this change, and instead make sure we document the need for correct `__sig` when building for wasm64.

Reply:
That seems suboptimal? ( https://github.com/emscripten-core/emscripten/issues/18200#issuecomment-1316836861 )

Reply:
Yes, there are probably ways we can optimize the current wasm64 wrappers/helpers.  But for now, the idea is that all pointers are considered i53 and are cast to Numbers when they enter JS, so JS library code, in general, doesn't have to worry about dealing bigints.

Perhaps, as you suggest, we could change that optimization such that function pointers are treated differently to data pointers?   Then only data pointers are considered i53 numbers while function pointers are not converted at the boundary and flow through JS are bigint?   But that would be something we should probably consider across the entire codebase.  

In my ideal world we could somehow tell the engine which arguments and return values are pointers and simply pass i53 number of JS without manifesting the bitint values at all.. but that would require some kind of spec change.